### PR TITLE
Ensuring sprite animation and sprite animation components get properly initialized

### LINKED ISSUE #227: Animation and AnimationComponent must ensure stepTime is informed
Is the a value for `stepTime` ins't informed, a NullPointerException can happen, we must add a `@required` to ensure that it is informed upon creation.

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Adding @required and asserts to ensure animations gets properly initialised, preventing null pointer exceptions.

Fixes #227

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected) (Kind of, I mean code would be broke already, anyway...)

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `v1.0.0`
- [x] This PR is targeted to merge into `v1.0.0` (not `master`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/components/sprite_animation_component.dart [hunk 1]
```diff
@@ -15,7 +15,7 @@ class SpriteAnimationComponent extends PositionComponent {
     Vector2 size,
     this.animation, {
     this.destroyOnFinish = false,
-  }) {
+  }) : assert(animation != null) {
```
Comment (spydon) [inline lib/components/sprite_animation_component.dart:18]: Can't you set `this.animation` as `@required` and it is automatically not nullable?

File: lib/sprite_animation.dart [hunk 2]
```diff
@@ -50,9 +51,9 @@ class SpriteAnimation {
   /// All frames have the same [stepTime].
   SpriteAnimation.spriteList(
     List<Sprite> sprites, {
-    double stepTime,
+    @required double stepTime,
     this.loop = true,
-  }) {
+  }) : assert(stepTime != null) {
```
Comment (spydon) [inline lib/sprite_animation.dart:56]: Both the assert and the @required shouldn't be needed right?

File: lib/components/sprite_animation_component.dart (hunk 1 above)
Comment (erickzanardo) [reply lib/components/sprite_animation_component.dart:18]: No :(, `@required` are only available to named parameters

File: lib/sprite_animation.dart (hunk 2 above)
Comment (erickzanardo) [reply lib/sprite_animation.dart:56]: they are separate things, `@required` means you need to informa the parameter on the call, but the person could inform null, so this is why I added both

Comment (spydon) [reply lib/sprite_animation.dart:56]: Ah, I misread an article about @required, but of course they can't know compile time whether the reference will be null. Approved. :)

**Review by spydon — APPROVED** [review]

File: lib/components/sprite_animation_component.dart (hunk 1 above)
Comment (luanpotter) [reply lib/components/sprite_animation_component.dart:18]: @required doesn't make it non-nullable, it makes it required (named parameters can be omitted). You can still explicitly pass in null. for real null safety we need to unable the new dart versio.

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**